src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -616,6 +616,34 @@ namespace cryptonote
     return true;
   }
   //-----------------------------------------------------------------------------------------------
+  std::pair<uint64_t, uint64_t> core::get_coinbase_tx_sum(const uint64_t start_offset, const uint64_t count)
+  {
+    std::list<block> blocks;
+    std::list<transaction> txs;
+    std::list<crypto::hash> missed_txs;
+    uint64_t coinbase_amount = 0;
+    uint64_t emission_amount = 0;
+    uint64_t total_fee_amount = 0;
+    uint64_t tx_fee_amount = 0;
+    this->get_blocks(start_offset, count, blocks);
```

Comment:
Yes, already_generated_coins does the emission portion. If you want the fees, you're right you'd have to go through the entire blockchain to scan.